### ISSUE TITLE
Internal error with namedtuple+classmethod

### ISSUE THREAD

**Issue by Daenyth** [python/mypy#3317]
Running off the latest master.

```
from typing import *

class Bar(NamedTuple):
    x: str

    @classmethod
    def new(cls, f: str) -> 'Bar':
        return Bar(x=f)
```

```
[mypy]
python_version = 3.6

check_untyped_defs = True
disallow_untyped_calls = True
follow_imports = True
incremental = True
strict_optional = True
warn_no_return = True
warn_return_any = True
warn_unused_ignores = True
```

```
foo.py:6: error: Invalid statement in NamedTuple definition; expected "field_name: field_type"
foo.py:6: error: INTERNAL ERROR -- please report a bug at https://github.com/python/mypy/issues version: 0.510-dev-09e7b124b82dae1207d9eac6805b64a98290015c
```

This might be fixed by #3081? I'm not sure how to install that to test

<details><summary>Stack trace is here</summary>

```
Traceback (most recent call last):
  File "/Users/daenyth/Curata/observer/.tox/mypy/bin/mypy", line 6, in <module>
    main(__file__)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/main.py", line 46, in main
    res = type_check_only(sources, bin_dir, options)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/main.py", line 93, in type_check_only
    options=options)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/build.py", line 188, in build
    graph = dispatch(sources, manager)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/build.py", line 1595, in dispatch
    process_graph(graph, manager)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/build.py", line 1838, in process_graph
    process_stale_scc(graph, scc, manager)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/build.py", line 1937, in process_stale_scc
    graph[id].type_check_first_pass()
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/build.py", line 1510, in type_check_first_pass
    self.type_checker.check_first_pass()
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 177, in check_first_pass
    self.accept(d)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 264, in accept
    stmt.accept(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/nodes.py", line 749, in accept
    return visitor.visit_class_def(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 1080, in visit_class_def
    self.accept(defn.defs)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 264, in accept
    stmt.accept(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/nodes.py", line 810, in accept
    return visitor.visit_block(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 1173, in visit_block
    self.accept(s)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 264, in accept
    stmt.accept(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/nodes.py", line 631, in accept
    return visitor.visit_decorator(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 2213, in visit_decorator
    e.func.accept(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/nodes.py", line 564, in accept
    return visitor.visit_func_def(self)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 513, in visit_func_def
    self.check_method_override(defn)
  File "/Users/daenyth/Curata/observer/.tox/mypy/lib/python3.6/site-packages/mypy/checker.py", line 945, in check_method_override
    for base in defn.info.mro[1:]:
TypeError: 'NoneType' object is not subscriptable
```

</details>

**Comment by Daenyth** [comment]
I figured out how to install it (deps line in tox needs `git+git://github.com/JelleZijlstra/mypy.git@ntmethods`)

#3081 does fix the crash

**Comment by JukkaL** [comment]
Keeping this open until until #3081 has been merged.

**Comment by ilevkivskyi** [comment]
@Daenyth By the way, do you still see the crash on master without ``incremental``?

**Comment by JukkaL** [comment]
I can repro without a config file and without incremental on master. Also verified that this doesn't happen on 0.501 so this is a regression and I'd like to get this fixed before release. Since we are in feature freeze, I'd rather have a specific fix for this issue instead of landing #3081. Anybody want to look at extracting the fix for this from #3081 (or implementing an independent fix)? 

^^ @JelleZijlstra @ilevkivskyi

**Comment by JelleZijlstra** [comment]
I have a (very hacky) quick fix; will send a pr shortly.
